62_CAD/DrawResourcesFiller.cpp

```diff
@@ -684,6 +787,33 @@ uint32_t DrawResourcesFiller::addLineStyle_Internal(const LineStyleInfo& lineSty
 	return currentLineStylesCount++;
 }
 
+uint32_t DrawResourcesFiller::addDTMSettings_Internal(const DTMSettingsInfo& dtmSettingsInfo, SIntendedSubmitInfo& intendedNextSubmit)
+{
+	DTMSettings dtmSettings;
+
+	// TODO: this needs to be redone.. what if submit happens after that line?
+	// we need to make sure somehow that function below will not submit, we need both outline and contour styles in GPU memory
+	dtmSettings.outlineLineStyleIdx = addLineStyle_SubmitIfNeeded(dtmSettingsInfo.outlineLineStyleInfo, intendedNextSubmit);
+	dtmSettings.contourLineStyleIdx = addLineStyle_SubmitIfNeeded(dtmSettingsInfo.contourLineStyleInfo, intendedNextSubmit);
+
+	DTMSettings* settingsArray = reinterpret_cast<DTMSettings*>(cpuDrawBuffers.dtmSettingsBuffer->getPointer());
+	for (uint32_t i = 0u; i < currentDTMSettingsCount; ++i)
+	{
+		const DTMSettings& itr = settingsArray[i];
+		if (itr == dtmSettings)
+			return i;
+	}
+
+	if (currentDTMSettingsCount >= maxDtmSettings)
```

Comment:
We should be extra careful and document more robustly. 
Auto submit can easily break with multiple _SubmitIfNeeded calls.
I'm sure I've stated explicitly somewhere in the comments that user shall not batch calls to  addMainObj.
Make sure you add the same comments to line styles and dtm settings (is shouldn't be called twice or more in a row and then used via index)